pkg/hostagent/port_darwin.go

```diff
@@ -149,8 +151,25 @@ func (plf *pseudoLoopbackForwarder) forward(ac *net.TCPConn) error {
 		return err
 	}
 	defer unixConn.Close()
-	bicopy.Bicopy(ac, unixConn, nil)
-	return nil
+
+	g, _ := errgroup.WithContext(context.Background())
+
+	g.Go(func() error {
+		_, err := io.Copy(unixConn, ac)
+		if errors.Is(err, io.EOF) {
+			err = nil
+		}
+		return errors.Join(err, unixConn.CloseRead(), ac.CloseWrite())
```

Comment:
Previous code logged errors when calling CloseRead and CloseWrite, but did not fail. This will return an error in this case.

Reply:
This is the same comment in triplicate, so resolving this one.

Reply:
This is not the same comment. You suggest to return error instead of logging them. Please explain why this is better and how the code calling the modified code handles the errors.

It can be useful to show logs with errors before and after the change.

Reply:
The caller of this function just logs the error. I have indicated this in the commit message.